pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/SegmentMetadata.java

```diff
@@ -98,11 +100,11 @@ public interface SegmentMetadata {
 
   String getEndOffset();
 
-  default Set<String> getAllColumns() {
+  default NavigableSet<String> getAllColumns() {
```

Comment:
Shall we directly use TreeSet instead of NavigableSet if there is no other implementation expected .

Reply:
It is an ordered set, but not a `TreeSet` because the key set of the `TreeMap` is not a `TreeSet`